xamarin/Xamarin.Forms
# Fixed PreferredStatusBarStyle not called on page renderers

### Description of Change ###

- Added override for ChildViewControllerForStatusBarStyle to ShellFlyoutRenderer to return renderer for current page
- Added override for ModalPresentationCapturesStatusBarAppearance to ModalWrapper so modal pages can also change style
- Added code for manual test

### Issues Resolved ### 

When using the Shell the `PreferredStatusBarStyle` method is not called in an iOS PageRenderer and therefore the colour of the status bar text cannot be changed on devices using latest OS.

- fixes #14246 

### API Changes ###

Added:
 - public override UIViewController ChildViewControllerForStatusBarStyle() -> ShellFlyoutRenderer.cs
 - public override bool ModalPresentationCapturesStatusBarAppearance => true; -> ModalWrapper.cs
 - Push Modal button -> StorePages.cs

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###
`PreferredStatusBarStyle` will begin to work when using page renderers and the Shell

### Testing Procedure ###

- Load Xamarin.Forms.Controls app in iOS
- Tap **SwapRoot - Store shell** located on the top list
- Notice the status bar text is black
- Tap List tab
- The status bar text is now white
- Tap Home tab
- The status bar text is now black again
- Scroll to bottom of the list
- Tap Push Model
- The status bar text is now white again (I should have changed the background colour of the page so it can be seen)

### PR Checklist ###
